lib/Doctrine/Migrations/Tools/Console/Command/ListCommand.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Tools\Console\Command;
+
+use DateTimeInterface;
+use Doctrine\Migrations\Metadata\AvailableMigration;
+use Doctrine\Migrations\Metadata\AvailableMigrationsList;
+use Doctrine\Migrations\Metadata\ExecutedMigration;
+use Doctrine\Migrations\Metadata\ExecutedMigrationsSet;
+use Doctrine\Migrations\Version\Version;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableCell;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use function array_map;
+use function array_merge;
+use function array_unique;
+use function uasort;
+
+/**
+ * The StatusCommand class is responsible for outputting a list of all available migrations and their status.
+ */
+class ListCommand extends DoctrineCommand
+{
+    /** @var string */
+    protected static $defaultName = 'migrations:list';
+
+    protected function configure() : void
+    {
+        $this
+            ->setAliases(['list-migrations'])
+            ->setDescription('Display a list of all available migrations and their status.')
+            ->setHelp(<<<EOT
+The <info>%command.name%</info> command outputs a list of all available migrations and their status:
+
+    <info>%command.full_name%</info>
+EOT
+            );
+
+        parent::configure();
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output) : ?int
+    {
+        $storage       = $this->getDependencyFactory()->getMetadataStorage();
+        $migrationRepo = $this->getDependencyFactory()->getMigrationRepository();
+
+        $availableMigrations = $migrationRepo->getMigrations();
+        $executedMigrations  = $storage->getExecutedMigrations();
+
+        $this->showVersions($availableMigrations, $executedMigrations, $output);
```

Comment:
How about
```suggestion
        $this->showVersions(
            $this->getDependencyFactory()->getMigrationRepository()->getMigrations(), // available migrations
            $this->getDependencyFactory()->getMetadataStorage()->getExecutedMigrations(), // executed migrations
            $output
        );
```

Fewer variables, no abbreviations, no assignments so no need to align those, no need to remember that we created a `$storage` variable until it is finally used, and you can even make sentences in the comments if you want to clarify things even more